.github/workflows/cppcheck.yml

```diff
@@ -0,0 +1,76 @@
+
+name: cppcheck
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  cppcheck:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+
+      - name: Set up dependencies
+        run: |
+          sudo apt update -y
+          sudo apt install -y --no-install-recommends --no-install-suggests \
+            build-essential \
+            cppcheck \
+            cmake \
+            ninja-build \
+            libssl-dev \
+            libcurl4-openssl-dev \
+            libprotobuf-dev \
+            protobuf-compiler \
+            libgmock-dev \
+            libgtest-dev \
+            libbenchmark-dev
+
+      - name: Prepare CMake
+        run: |
+          mkdir build && cd build
+          CC="clang" CXX="clang++" cmake ..
+
+      - name: Run cppcheck
+        run: |
+          cppcheck \
```

Comment:
Please add another statement, to run `cppcheck --version`, so we can see the actual version used in CI logs.
This will help debugging.

Reply:
Done in CI only PR #3151